### PR TITLE
CSS module: Conditional rules

### PR DESCRIPTION
**Pull request by estelle**
New (rewrite) module landing page.
Rewrite the Feature Queries guide.

### CODE REVIEW

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md [hunk 1]
```diff
@@ -53,10 +76,12 @@ You may need to test support for more than one property in your feature query. T
 }
 ```
 
-For example, if the CSS you want to run requires that the browser supports CSS Shapes and CSS Grid, you could create a rule which checks for both of these things. The following rule will only return true if both `shape-outside: circle()` and `display: grid` are supported by the browser.
+For example, if the CSS you want to run requires that the browser supports CSS Shapes and CSS Grid, you could create a rule that checks for both of these things. The following rule will only return true if `shape-outside: circle()` and `display: grid` are both supported by the browser.
```
Comment (chrisdavidmills) [inline files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md:79]:
```suggestion
For example, if the CSS you want to run requires that the browser supports CSS Shapes and CSS Grid, you could create a rule that tests browser support for both of these features. The following rule will only return true if `shape-outside: circle()` and `display: grid` are both supported by the browser.
```

Comment (chrisdavidmills) [reply files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md:79]: Also, link "CSS Shapes" and "CSS Grid" to the relevant module landing pages?

Comment (chrisdavidmills) [reply files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md:79]: Also, the example is again illogical — red should indicate lack of support, not support. I'll not point this out for future examples, but make sure you keep it in mind as you look at the ones below this one and decide what to do about them.

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md [hunk 2]
```diff
@@ -53,10 +76,12 @@ You may need to test support for more than one property in your feature query. T
...
 
 {{EmbedGHLiveSample("css-examples/feature-queries/and.html", '100%', 600)}}
 
+## Testing for at least one of multiple features
+
 You can also use `or`, if one property out of a selection could match to enable the CSS you want to use:
```
Comment (chrisdavidmills) [inline files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md:85]:
```suggestion
You can also use `or` to apply CSS only if one or more declarations are supported:
```

Comment (chrisdavidmills) [reply files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md:85]: Also, maybe update the below example to test for a `-webkit-` prefixed property of some kind? Testing for `-ms-grid` is completely pointless in the modern browser landscape.

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md [hunk 3]
```diff
@@ -71,47 +96,19 @@ This can be particularly useful if a feature is vendor prefixed, as you can test
 
 ## Limitations of feature queries
 
-The `@supports` rule tests to see if the browser can parse one or more property/value pairs, and therefore if it claims to support the feature(s). If the property and value pair is understood by the browser it returns a positive response. Therefore feature queries cannot be used to check if a browser supports a thing properly, and without bugs!
+The `@supports` rule tests to see if the browser can parse one or more property/value pairs, and therefore if it claims to support the feature(s). If the property and value pair is understood by the browser it returns a positive response. Feature queries checks that the declaration would be considered valid by the browser, but can not be used to check if a browser supports a thing properly, and without bugs. Feature queries cannot test for _partial implementations_.
```
Comment (chrisdavidmills) [inline files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md:99]:
```suggestion
The `@supports` rule tests to see if browsers can parse one or more property/value pairs, and therefore if they claim to support the associated feature(s). If the property/value pairs are understood by a browser it returns a positive response. Feature queries check that declarations are considered valid by a browser, but can't be used to check if it supports a feature properly without bugs or spec violations. Feature queries cannot test for _partial implementations_.
```

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md [hunk 4]
```diff
@@ -71,47 +96,19 @@ This can be particularly useful if a feature is vendor prefixed, as you can test
...
-
-In the above scenario, it doesn't matter that IE11 doesn't support feature queries or CSS Grid — it would get the floated version anyway, which is then overwritten by browsers that do support grid.
-
-An alternate way to write the above code is to wrap all of the grid code in a feature query as follows.
-
-{{EmbedGHLiveSample("css-examples/feature-queries/step4.html", '100%', 900)}}
+## Summary
 
-This may mean you have a little more code but comes with the benefit of being able to test the fallback by misspelling the property or value name. You can try this in the live example above by changing `display: grid` in the `@supports` rule to `display: grip` or similar.
+Feature queries are a useful tool when progressively enhancing a site. As we've seen, they enable you to provide a good solution for all browsers, and an enhanced solution for those browsers that support newer properties and values.
```
Comment (chrisdavidmills) [inline files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md:103]:
```suggestion
Feature queries are a useful tool for progressively enhancing a site. They enable you to provide a good solution for all browsers, and an enhanced solution for browsers that support newer properties and values.
```

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md [hunk 5]
```diff
@@ -71,47 +96,19 @@ This can be particularly useful if a feature is vendor prefixed, as you can test
...
 
-## Summary
+You don't need to use feature queries to start using new CSS features; CSS error handling means the browser simply ignores CSS it does not yet recognize. However, feature queries are a useful alternative to fallback declarations, and enable writing code once that can eventually be supported everywhere.
 
-Feature Queries can help you start to use newer features by enhancing a simpler display of the site used for older browsers. As you can wrap up the CSS for supporting browsers, you do not run the risk of styles used for the fallback display leaking through, as shown in our grid example above.
+Feature queries are not limited to property-value pairs. You can also check support for [`font-tech()`](/en-US/docs/Web/CSS/@supports#font-tech), [`font-format()`](/en-US/docs/Web/CSS/@supports#font-format), and [`selector()`](/en-US/docs/Web/CSS/@supports#function_syntax).
```
Comment (chrisdavidmills) [inline files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md:107]: I'd update the link text in these links to describe in words what is being tested for, not just list the functions used to write the necessary support queries, so for example "font technologies", "font formats", and "selectors". When I first read this I went, for example, "hrm, I wonder what the selector() function is and why I'd want to test for that?" It took me a few seconds to realize that you just meant selectors.

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md [hunk 6]
```diff
@@ -71,47 +96,19 @@ This can be particularly useful if a feature is vendor prefixed, as you can test
...
-
-What we need is a way to remove the width if `display: grid` is supported. This is exactly the situation feature queries solve. We can set the `width` back to `auto` if grid is supported.
-
-{{EmbedGHLiveSample("css-examples/feature-queries/step3.html", '100%', 900)}}
-
-In the above scenario, it doesn't matter that IE11 doesn't support feature queries or CSS Grid — it would get the floated version anyway, which is then overwritten by browsers that do support grid.
-
-An alternate way to write the above code is to wrap all of the grid code in a feature query as follows.
-
-{{EmbedGHLiveSample("css-examples/feature-queries/step4.html", '100%', 900)}}
```
Comment (chrisdavidmills) [inline files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md:104]: I know this example is out of date, hence you getting rid of it. However, it does feel like a bit of a shame that the article no longer has a "real" example in it. Could you maybe add back in a simpler one showing usage of a feature query, or link to some examples elsewhere?

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
@estelle more great work, thanks for this. I've left you some comments and suggestions, but nothing too serious.

File: files/en-us/web/css/css_conditional_rules/index.md [hunk 7]
```diff
@@ -2,30 +2,123 @@
...
+
+The first CSS conditional rules were [media types](/en-US/docs/Web/CSS/@media#media_types) specifying the intended destination medium for the linked styles. These were set as the value of the HTML {{HTMLElement("link")}} and {{HTMLElement("style")}} elements' `media` attributes or as a comma-separated list of media types within an {{cssxref("@import")}} statement and {{cssxref("@import")}} at-rule. The ability to conditionally apply CSS rules has been greatly expanded since the CSS 2.1 and HTML 4.01 implementations that limited conditional queries to a few media types.
+
+CSS conditional rules now include feature queries; the `@supports` at-rule enables targeting CSS styles based on a user-agent's CSS capabilities. Additional conditions include which selector, font-formats, and font-techs are supported.
+
+The CSS conditional rules module also expanded `@media` to enable nesting at-rules with the [CSS media queries](/en-US/docs/Web/CSS/CSS_media_queries) removing unused media types while adding many media features and conditions that can be targeted.
```
[the remark this answers is not in the document]
Comment (estelle) [reply files/en-us/web/css/css_conditional_rules/index.md:19]: media is introduced in this spec but really covered in the media queries spec. When discussing history, @media should be listed first. In the content of this page, @supports is more relevant than @media content. This spec expands @media to include nesting, but actual media types are in a different spec, so its an aside compared to supports

File: files/en-us/web/css/css_conditional_rules/index.md [hunk 8]
```diff
@@ -2,30 +2,123 @@
...
+- [`<media-query>`](/en-US/docs/Web/CSS/CSS_media_queries/Using_media_queries#syntax)
+- [`<supports-condition>`](/en-US/docs/Web/CSS/@import#importing_css_rules_conditional_on_feature_support)
+- `<supports-feature>` (see [`supports()`](/en-US/docs/Web/CSS/@import#supports-condition))
+
+### Interfaces
+
+- {{domxref("CSSConditionRule")}}
+- {{domxref("CSSMediaRule")}}
+- {{domxref("CSSSupportsRule")}}
+- {{domxref("CSS.supports_static")}} method
```
Comment (estelle) [inline files/en-us/web/css/css_conditional_rules/index.md:54]:
```suggestion
- {{domxref("CSS.supports_static", "supports()")}} method
```

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md (hunk 2 above)
Comment (estelle) [reply files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md]: see https://github.com/mdn/css-examples/pull/166

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md (hunk 6 above)
Comment (estelle) [reply files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md:104]: added an accent color example with both regular and "not"

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md (hunk 5 above)
Comment (estelle) [reply files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md]: moved up into a "## Additional feature query options" section and added a faux example.

File: files/en-us/web/css/css_conditional_rules/index.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/css/css_conditional_rules/index.md:19]: OK, fair point.

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md [hunk 9]
```diff
@@ -6,28 +6,51 @@ page-type: guide
...
+
+```css
+@supports (color: red) {
+  CSS rules to apply
+}
+
+@import `/css/styles.css` supports(color: red);
 ```
 
 If you want to check if a browser supports the `row-gap` property, for example, you would write the following feature query. It doesn't matter which value you use in a lot of cases: if all you want is to check that the browser supports this property, then any valid value will do.
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md]: @estelle looks like this one still needs addressing.

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md (hunk 1 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md]: @estelle I see you've updated some of these examples, which is great, but red indicating support is still a bit odd. Wouldn't green and dashed border for support be better?

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md (hunk 2 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md]: @estelle Nice one - this example is now much more logical. However, the example has a bug in it and needs an update. As per the first note on https://developer.mozilla.org/en-US/docs/Web/CSS/font-smooth, `-webkit-font-smoothing` takes different values to the non-prefixed `font-smooth` property.

The example works if you swap out `-webkit-font-smoothing: always` for `-webkit-font-smoothing: antialiased`.

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md (hunk 5 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md]: Great! This works much better now.

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md [hunk 10]
```diff
@@ -69,49 +103,67 @@ This can be particularly useful if a feature is vendor prefixed, as you can test
...
+```
 
-Let's say we want to create a layout of three boxes in a row, and ideally we would like to use [CSS Grid Layout](/en-US/docs/Web/CSS/CSS_grid_layout). However, we would like to have a layout for older browsers using floats. We can start by creating that floated layout with the following code, which gives us three columns.
+## Examples
 
-{{EmbedGHLiveSample("css-examples/feature-queries/step1.html", '100%', 900)}}
+### Browser support test
 
-When browsers don't understand a CSS property or value, they ignore it. So we could start enhancing our layout by using CSS Grid. Browsers that do not support grid will ignore the `grid` value of the `display` property. Once a floated item becomes a grid item, the float is removed — something you can read more about in [Supporting Older Browsers](/en-US/docs/Learn/CSS/CSS_layout/Supporting_Older_Browsers). Therefore the grid version should just overwrite the floated one.
+In this example, we check if the browser supports {{cssxref("system-color")}}, hiding support with `diplay: none` if the color type is not supported.
```
Comment (chrisdavidmills) [inline files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md:121]:
```suggestion
In this example, we check if the browser supports the `AccentColor` {{cssxref("system-color")}}, hiding support with `display: none` if the color type is not supported.
```

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
Thanks for all the additional work on this one, @estelle; this is now looking much better.

I had a few additional comments on the new text you added, and I think the examples still need some work. Much closer now, however.

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md (hunk 2 above)
Comment (estelle) [reply files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md]: fixed in https://github.com/mdn/css-examples/pull/167

**Comment by estelle** [comment]
Thanks @chrisdavidmills !

File: files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md (hunk 2 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/css_conditional_rules/using_feature_queries/index.md]: Perfect!

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
@estelle IMO, the article wording is perfect now. I just made a small tweak to the `AccentColor` example as some of the wording didn't seem to reflect what the example was doing. You can check my change in https://github.com/mdn/content/pull/31387/commits/df55654952f61cc9f4b9482706f11117ae065364 to make sure you are happy with it.

I still have concerns over the embedded GH examples though — it looks like you've not addressed these comments:

- https://github.com/mdn/content/pull/31387#discussion_r1440312589
- https://github.com/mdn/content/pull/31387#discussion_r1440315661

**Comment by estelle** [comment]
It would help if i saved files before committing them ;)
Fixed in https://github.com/mdn/css-examples/pull/168

**Comment by chrisdavidmills** [comment]
@estelle inching closer. The example at https://mdn.github.io/css-examples/feature-queries/simple.html (the one embedded below the paragraph that starts "As another example, if you want to check if a browser supports...") is still using red to indicate support.

**Comment by estelle** [comment]
The files in other repos have been updated.

**Review by chrisdavidmills — APPROVED** [review]
@estelle perfect, thanks for all the great work here. I'll get this merged.

**Pull request merged**